modules/ding_availability/js/ding_availability.js

```diff
@@ -9,79 +9,80 @@
-    if (settings.hasOwnProperty('ding_availability')) {
-      $.each(settings.ding_availability, function (id, entity_ids) {
-        $.each(entity_ids, function (index, entity_id) {
-          if (Drupal.DADB[entity_id] === undefined) {
-            Drupal.DADB[entity_id] = null;
-            ids.push(entity_id);
-            html_ids.push(id);
-          }
-        });
-      });
-    }
+  Drupal.behaviors.dingAvailabilityAttach = {
+    attach: function(context, settings) {
+      var ids = [];
+      var html_ids = [];
 
-    // If there's any reservation buttons, switch to holdings. We have
-    // periodicals that's a bit off an odd one in that they can both
-    // have issues, which means that the reservation button for the
-    // main object should be disabled, or not have any issues, which
-    // means that it should be left alone. So we need to fetch full
-    // holdings in order to determine whether the material is a
-    // periodical. This is a bit of a hack, but it's the quickest way
-    // of fixing the problem right now.
-    if ($('.reserve-button').size() > 0) {
-      settings.ding_availability_mode = 'holdings';
-    }
+      // Loop through the materials given in the settings and collect
+      // HTML ids and entity_ids.
+      if (settings.hasOwnProperty('ding_availability')) {
+        $.each(settings.ding_availability, function (id, entity_ids) {
+          $.each(entity_ids, function (index, entity_id) {
+            if (Drupal.DADB[entity_id] === undefined) {
+              Drupal.DADB[entity_id] = null;
+              ids.push(entity_id);
+              html_ids.push(id);
+            }
+          });
+        });
+      }
 
-    $.each(html_ids, function (index, id) {
-      $('#' + id).addClass('pending');
-    });
+      // If there's any reservation buttons, switch to holdings. We have
+      // periodicals that's a bit off an odd one in that they can both
+      // have issues, which means that the reservation button for the
+      // main object should be disabled, or not have any issues, which
+      // means that it should be left alone. So we need to fetch full
+      // holdings in order to determine whether the material is a
+      // periodical. This is a bit of a hack, but it's the quickest way
+      // of fixing the problem right now.
+      if ($('.reserve-button').size() > 0) {
+        settings.ding_availability_mode = 'holdings';
+      }
 
-    // Fetch availability.
-    if (ids.length > 0) {
-      var mode = settings.ding_availability_mode ? settings.ding_availability_mode : 'items';
-      var path = settings.basePath + 'ding_availability/' + mode + '/' + ids.join(',');
-      $.ajax({
-        dataType: "json",
-        url: path,
-        success: function (data) {
-          $.each(data, function (id, item) {
-            // Update cache.
-            Drupal.DADB[id] = item;
-          });
+      $.each(html_ids, function (index, id) {
+        $('#' + id).addClass('pending');
```

Comment:
This could be limited to the `context` of the behavior, but it might not make sense since we are locating by id anyhow.